dev_support/ecto/repo.ex

```diff
@@ -12,5 +12,23 @@ if FunWithFlags.Config.persist_in_ecto? do
     end)
 
     use Ecto.Repo, otp_app: :fun_with_flags, adapter: @variant
+    
+    # for testing setups that use multi-tenancy using foreign keys
+    # as described in the Ecto docs:
+    # https://hexdocs.pm/ecto/multi-tenancy-with-foreign-keys.html
+    #
+    # FunWithFlags sets the custom query option `:fun_with_flags` to
+    # true to allow such setups to detect queries originating from
+    # FunWithFlags
+    @impl true
+    def prepare_query(_operation, query, opts) do
+      cond do
+        opts[:schema_migration] || opts[:fun_with_flags] ->
+          {query, opts}
+
+        true ->
+          raise "expected fun_with_flags query option to be set"
+      end
+    end
```

Comment:
I wouldn't add this here.

Reply:
I mean, it's convenient to have it here to inspect the options, but maybe you can comment it out.

Reply:
So the reason I added this is the following: the commit introduces a custom ecto query option that should be set for all queries originating from the library. Instead of adding a test for all calls, I added this to instead raise if a test executes a query that does not has the flag set. Therefore any code changes that forget to set the query option will lead to failing tests.

Reply:
So just to follow up: do you still want this to be changed? If yes, do you want the whole `prepare_query/3` removed?
I think it should at least log a warning if an unexpected query executes without the query option, but raising and therefore failing tests is still the solution I like the most. What do you think?

Reply:
Yes, sorry. What you wrote makes sense. Let's keep this.